### PR TITLE
CONCF 925 | As an admin I want to save my changes so I can manage my items

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/CONCF-925

### CODE REVIEW

File: front/scripts/main/mixins/ObjectUtilitiesMixin.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+App.ObjectUtilitiesMixin = Em.Mixin.create({
+	toJSON: function() {
+		var value: any,
+			keys: any[] = [];
+		for (var key in this) {
+			if (this.hasOwnProperty(key)) {
+				value = this[key];
+				if (value === 'toString') {
```
Comment (bognix) [inline front/scripts/main/mixins/ObjectUtilitiesMixin.ts]: isn't Ember.typeOf(toString) function too?

Comment (tomasznapieralski) [inline front/scripts/main/mixins/ObjectUtilitiesMixin.ts]: http://stackoverflow.com/questions/8669340/ember-model-to-json

File: front/scripts/main/models/CuratedContentEditorModel.ts [hunk 2]
```diff
@@ -101,7 +144,7 @@ App.CuratedContentEditorModel.reopenClass({
 	},
 
 	addItem(parent: CuratedContentEditorItemModel, newItem: CuratedContentEditorItemModel): void {
-		parent.items.push(newItem);
+		parent.items.push(newItem.toJSON());
```
Comment (hakubo) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: How about just using
```newItem.getProperties(...)```

File: front/scripts/main/models/CuratedContentEditorModel.ts [hunk 3]
```diff
@@ -45,6 +68,26 @@ App.CuratedContentEditorModel.reopenClass({
...
+	 * @desc Convert CuratedContentEditorItemModel to array known for CuratedContent API
+	 *
+	 * @param model CuratedContentEditorItemModel
+	 * @returns {Array}
+	 */
+	prepareDataForSave(model: CuratedContentEditorItemModel): string[] {
+		var data: string[] = [];
+		data.push(model.featured);
+
+		model.curated.items.forEach((section: string): void => {
```
Comment (hakubo) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: `data.concat(model.featured, model.curated.items, model.optional);`

File: front/scripts/main/models/CuratedContentEditorModel.ts (hunk 2 above)
Comment (rogatty) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: @hakubo: this is exactly what `toJSON` does, see https://github.com/Wikia/mercury/pull/1143/files#diff-358ebae2dcf1d2b3c177c5cf37441b3fR18.

One thing that it adds is being smart about the list of properties to get. Do we really want to have another static list of them?

**Comment by vforge** [comment]
Those changes are already implemented in https://github.com/Wikia/mercury/pull/1148/files
Closing...

**Pull request closed without merging**